### PR TITLE
media-libs/opencollada: Remove 1.2.2_p20150207 and version bump to 1.6.36

### PR DESCRIPTION
**Pull request by dracwyrm**
media-libs/opencollada: Remove 1.2.2_p20150207:
This version has several security vulnerabilities which are fixed in
newer versions.

media-libs/opencollada: Version bump to 1.6.36
* General bug fix bump with no added features.

@gentoo/proxy-maint

### CODE REVIEW

File: media-libs/opencollada/opencollada-1.6.36.ebuild [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+	"${FILESDIR}"/${PN}-0_p864-expat.patch
+	"${FILESDIR}"/${PN}-1.2.2-soversion.patch
+	"${FILESDIR}"/${PN}-1.2.2-no-undefined.patch
+	"${FILESDIR}"/${PN}-1.2.2-libdir.patch
+)
+
+src_prepare() {
+	edos2unix CMakeLists.txt
+
+	default
```
Comment (SoapGentoo) [inline media-libs/opencollada/opencollada-1.6.36.ebuild]: don't use `default` with cmake-utils, only ever use `cmake-utils_src_prepare`

File: media-libs/opencollada/opencollada-1.6.36.ebuild [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+	media-libs/lib3ds
+	sys-libs/zlib
+	expat? ( dev-libs/expat )
+	!expat? ( dev-libs/libxml2 )"
+DEPEND="${RDEPEND}
+	virtual/pkgconfig"
+
+S="${WORKDIR}"/OpenCOLLADA-${PV}
+# This is needed or you get an error on install
+BUILD_DIR="${S}"/build
```
Comment (SoapGentoo) [inline media-libs/opencollada/opencollada-1.6.36.ebuild]: can't you just fix the build system instead?

File: media-libs/opencollada/opencollada-1.6.36.ebuild [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+	rm -v Makefile scripts/{unixbuild.sh,vcproj2cmake.rb} || die
+	find "${S}" -name SConscript -delete || die
+}
+
+src_configure() {
+	local mycmakeargs=(
+		-DUSE_SHARED=ON
+		-DUSE_STATIC=$(usex static-libs ON OFF)
+		-DUSE_EXPAT=$(usex expat ON OFF)
+		-DUSE_LIBXML=$(usex !expat ON OFF)
```
Comment (SoapGentoo) [inline media-libs/opencollada/opencollada-1.6.36.ebuild]: if possible remove the `ON`s and `OFF`s, they make reading harder, I have to first check if they aren't inverted etc. Short concise `usex` make reading much easier

File: media-libs/opencollada/files/opencollada-build-fixes-v1.patch [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+@@ -270,10 +266,10 @@ else ()  # if pcre not found building it
+ endif ()
+ 
+ # building required libs
++add_subdirectory(${EXTERNAL_LIBRARIES}/UTF)                                    
++add_subdirectory(${EXTERNAL_LIBRARIES}/MathMLSolver)
+ add_subdirectory(common/libftoa)
+ add_subdirectory(common/libBuffer)
+-add_subdirectory(${EXTERNAL_LIBRARIES}/UTF)
+-add_subdirectory(${EXTERNAL_LIBRARIES}/MathMLSolver)
```
Comment (SoapGentoo) [inline media-libs/opencollada/files/opencollada-build-fixes-v1.patch]: does this churn add any value? seems kinda pointless to me

File: media-libs/opencollada/files/opencollada-build-fixes-v1.patch [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+@@ -190,9 +187,9 @@ set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS}
+ #-----------------------------------------------------------------------------
+ # Install vars
+ 
+-set(OPENCOLLADA_INST_INCLUDE ${CMAKE_INSTALL_PREFIX}/include/opencollada)
+-set(OPENCOLLADA_INST_LIBRARY ${CMAKE_INSTALL_PREFIX}/lib/opencollada)
+-set(OPENCOLLADA_INST_CMAKECONFIG ${OPENCOLLADA_INST_LIBRARY}/cmake)
++set(OPENCOLLADA_INST_INCLUDE ${CMAKE_INSTALL_FULL_INCLUDEDIR}/opencollada)
++set(OPENCOLLADA_INST_LIBRARY ${CMAKE_INSTALL_FULL_LIBDIR}/opencollada)
++set(OPENCOLLADA_INST_CMAKECONFIG ${CMAKE_INSTALL_FULL_LIBDIR}/opencollada/cmake)
```
Comment (SoapGentoo) [inline media-libs/opencollada/files/opencollada-build-fixes-v1.patch]: this is tricky. If you use `OPENCOLLADA_INST_CMAKECONFIG` with `INSTALL()`, this is generally discouraged, as `INSTALL()` should be used with relative, not absolute paths. If you use it to pass into `CPPFLAGS` for incorporating into `.cpp` files, it's ok.

Comment (dracwyrm) [inline media-libs/opencollada/files/opencollada-build-fixes-v1.patch]: I can ask upstream why they wanted full paths in the first place.

File: media-libs/opencollada/files/opencollada-build-fixes-v1.patch (hunk 4 above)
Comment (dracwyrm) [inline media-libs/opencollada/files/opencollada-build-fixes-v1.patch]: This has to do with ordering for this block:
```
-set(TARGET_LIBS ftoa)
+set(TARGET_LIBS
+	ftoa
+	UTF
+)
```

Both those are modifications are based on an OpenSUSE patch.

Comment (dracwyrm) [inline media-libs/opencollada/files/opencollada-build-fixes-v1.patch]: This one to be precise: http://dpaste.com/127RRF0

**Pull request closed without merging**